src/material-experimental/mdc-dialog/dialog-content-directives.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnChanges,
+  OnInit,
+  Optional,
+  SimpleChanges,
+} from '@angular/core';
+
+import {MatDialog} from './dialog';
+import {MatDialogRef} from './dialog-ref';
+
+/** Counter used to generate unique IDs for dialog elements. */
+let dialogElementUid = 0;
+
+/**
+ * Button that will close the current dialog.
+ */
+@Directive({
+  selector: '[mat-dialog-close], [matDialogClose]',
+  exportAs: 'matDialogClose',
+  host: {
+    '(click)': 'dialogRef.close(dialogResult)',
+    '[attr.aria-label]': 'ariaLabel || null',
+    '[attr.type]': 'type',
```

Comment:
Just `[type]`?

Reply:
This is from the non-MDC dialog too. I think we can't switch that one since we don't necessarily know whether the directive is placed on a button element.